src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -1171,16 +1171,15 @@ CleanUpSessions(DistributedExecution *execution)
 				elog(WARNING, "unexpected transaction state at the end of execution: %d",
 					 transactionState);
 			}
+
+			/* get ready for the next executions if we need use the same connection */
+			connection->waitFlags = WL_SOCKET_READABLE | WL_SOCKET_WRITEABLE;
 		}
 		else
 		{
 			elog(WARNING, "unexpected connection state at the end of execution: %d",
```

Comment:
This is unreachable since we never set connection state to `MULTI_CONNECTION_INITIAL`. I'd prefer to replace with an `Assert()`, but maybe we should postpone that to another PR.